### DISCUSSION TITLE
Support array list of extensions for custom data file formats

### DISCUSSION THREAD

**Discussion by Snapstromegon** [11ty/eleventy#4135] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

Providing multiple file extensions like `yml` and `yaml` as a comma separated list is often fine, but especially when generating these extension mappings programmatically a list would be easier to use.
This is related to #2381.

### Describe the solution you'd like

I would want this to work too:

```js
eleventyConfig.addDataExtension(["yml", "yaml"], (contents, filePath) => {
  // …
});
```

### Additional context

Like always I'd be willing to contribute a PR resolving this.

**Comment by zachleat** [comment]
Code for this lives here: https://github.com/11ty/eleventy/blob/a9ee377494a68934a8233559e3986839bdff5d5e/src/UserConfig.js#L956

Tests can be added here: https://github.com/11ty/eleventy/blob/a9ee377494a68934a8233559e3986839bdff5d5e/test/UserDataExtensionsTest.js
